src/options/config.rs

```diff
@@ -0,0 +1,538 @@
+    Env(PathBuf), // $EZA_CONFIG_DIR
+}
+
+trait FromOverride<T>: Sized {
+    fn from(value: T, default: Self) -> Self;
+}
+
+impl<S, T> FromOverride<Option<S>> for Option<T>
+where
+    T: FromOverride<S> + Default,
+{
+    fn from(value: Option<S>, default: Option<T>) -> Option<T> {
+        match (value, default) {
+            (Some(value), Some(default)) => Some(FromOverride::from(value, default)),
+            (Some(value), None) => Some(FromOverride::from(value, T::default())),
+            (None, Some(default)) => Some(default),
+            (None, None) => None,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Serialize, Deserialize, Default)]
+pub struct StyleOverride {
+    /// The style's foreground color, if it has one.
+    pub foreground: Option<Color>,
+
+    /// The style's background color, if it has one.
+    pub background: Option<Color>,
+
+    /// Whether this style is bold.
+    pub is_bold: Option<bool>,
+
+    /// Whether this style is dimmed.
+    pub is_dimmed: Option<bool>,
+
+    /// Whether this style is italic.
+    pub is_italic: Option<bool>,
+
+    /// Whether this style is underlined.
+    pub is_underline: Option<bool>,
+
+    /// Whether this style is blinking.
+    pub is_blink: Option<bool>,
+
+    /// Whether this style has reverse colors.
+    pub is_reverse: Option<bool>,
+
+    /// Whether this style is hidden.
+    pub is_hidden: Option<bool>,
+
+    /// Whether this style is struckthrough.
+    pub is_strikethrough: Option<bool>,
+
+    /// Wether this style is always displayed starting with a reset code to clear any remaining style artifacts
+    pub prefix_with_reset: Option<bool>,
+}
+
+impl FromOverride<StyleOverride> for Style {
+    fn from(value: StyleOverride, default: Self) -> Self {
+        let mut style = default;
+        if value.foreground.is_some() {
+            style.foreground = value.foreground;
+        }
+        if value.background.is_some() {
+            style.background = value.background;
+        }
+        if let Some(bold) = value.is_bold {
+            style.is_bold = bold;
+        }
+        if let Some(dimmed) = value.is_dimmed {
+            style.is_dimmed = dimmed;
+        }
+        if let Some(italic) = value.is_italic {
+            style.is_italic = italic;
+        }
+        if let Some(underline) = value.is_underline {
+            style.is_underline = underline;
+        }
+        if let Some(blink) = value.is_blink {
+            style.is_blink = blink;
+        }
+        if let Some(reverse) = value.is_reverse {
+            style.is_reverse = reverse;
+        }
+        if let Some(hidden) = value.is_hidden {
+            style.is_hidden = hidden;
+        }
+        if let Some(strikethrough) = value.is_strikethrough {
+            style.is_strikethrough = strikethrough;
+        }
+        if let Some(reset) = value.prefix_with_reset {
+            style.prefix_with_reset = reset;
```

Comment:
I think using some `unwrap_or` or `unwrap_or_default` could make the code more readable. What do you think ?

Reply:
I'm not sure but I don't think that was an option. that is how it ended up outputting the wrong escape codes previously

Reply:
We could do something like:
```
style.is_bold = value.is_bold.unwrap_or(style.is_bold);
```
I did consider that originally but decided against it because even though it is shorter it at least in my opinion actually makes the logic less clear, because what we are doing is
```
a = b.unwrap_or(a);
```
which assigns to `a` in both cases and in the `None` case assigns `a` to itself, which is visible in this simplified version, but with `style.is_bold` and `value.is_bold` you might easily overlook what is what. The `if let` clearly captures that we only change something when the thing in `value` is `Some`.

But we can change this if you both think the `unwrap_or` is more intuitive.

Reply:
> which assigns to a in both cases and in the None case assigns a to itself, which is visible in this simplified version, but with style.is_bold and value.is_bold you might easily overlook what is what. The if let clearly captures that we only change something when the thing in value is Some

I didn't thought of it like that but you got a point here

Reply:
Should we mark this as resolved then?